File: backend/start_server.py
#!/usr/bin/env python3
"""
Simple script to start the FastAPI server
"""

import sys
import os
import subprocess

def check_dependencies():
    """Check if required packages are installed"""
    try:
        import uvicorn
        import fastapi
        print("✓ FastAPI and Uvicorn are installed")
        return True
    except ImportError as e:
        print(f"✗ Missing dependencies: {e}")
        return False

def install_dependencies():
    """Install required dependencies"""
    print("Installing dependencies...")
    try:
        subprocess.check_call([sys.executable, "-m", "pip", "install", "-r", "requirements.txt"])
        print("✓ Dependencies installed successfully")
        return True
    except subprocess.CalledProcessError:
        print("✗ Failed to install dependencies")
        return False

def start_server():
    """Start the FastAPI server"""
    try:
        import uvicorn
        print("Starting FastAPI server...")
        print("Server will be available at: http://localhost:8000")
        print("API Documentation: http://localhost:8000/docs")
        print("Press Ctrl+C to stop the server")
        
        uvicorn.run(
            "server:app",
            host="127.0.0.1",
            port=8000,
            reload=True,
            log_level="info"
        )
    except Exception as e:
        print(f"✗ Failed to start server: {e}")
        return False

def main():
    print("=== Plagiarism Control Backend Server ===")
    print()
    
    # Check if dependencies are installed
    if not check_dependencies():
        print("Installing missing dependencies...")
        if not install_dependencies():
            print("Please install dependencies manually:")
            print("pip install -r requirements.txt")
            sys.exit(1)
    
    # Start the server
    start_server()

if __name__ == "__main__":
    main()

File: backend/start-server.js
#!/usr/bin/env node

/**
 * Simple script to start the Node.js server
 */

const { spawn } = require('child_process');
const fs = require('fs');
const path = require('path');

function checkDependencies() {
  console.log('Checking dependencies...');
  
  const packageJsonPath = path.join(__dirname, 'package.json');
  const nodeModulesPath = path.join(__dirname, 'node_modules');
  
  if (!fs.existsSync(packageJsonPath)) {
    console.log('✗ package.json not found');
    return false;
  }
  
  if (!fs.existsSync(nodeModulesPath)) {
    console.log('✗ node_modules not found');
    return false;
  }
  
  console.log('✓ Dependencies appear to be installed');
  return true;
}

function installDependencies() {
  console.log('Installing dependencies...');
  
  return new Promise((resolve, reject) => {
    const npm = spawn('npm', ['install'], {
      stdio: 'inherit',
      cwd: __dirname
    });
    
    npm.on('close', (code) => {
      if (code === 0) {
        console.log('✓ Dependencies installed successfully');
        resolve(true);
      } else {
        console.log('✗ Failed to install dependencies');
        reject(false);
      }
    });
    
    npm.on('error', (error) => {
      console.log('✗ Failed to run npm install:', error.message);
      reject(false);
    });
  });
}

function startServer() {
  console.log('Starting Node.js server...');
  console.log('Server will be available at: http://localhost:8000');
  console.log('API Base URL: http://localhost:8000/api');
  console.log('Press Ctrl+C to stop the server');
  console.log('');
  
  const server = spawn('node', ['server.js'], {
    stdio: 'inherit',
    cwd: __dirname
  });
  
  server.on('error', (error) => {
    console.log('✗ Failed to start server:', error.message);
    process.exit(1);
  });
  
  // Handle graceful shutdown
  process.on('SIGINT', () => {
    console.log('\n🛑 Shutting down server...');
    server.kill('SIGINT');
    process.exit(0);
  });
  
  process.on('SIGTERM', () => {
    console.log('\n🛑 Shutting down server...');
    server.kill('SIGTERM');
    process.exit(0);
  });
}

async function main() {
  console.log('=== Plagiarism Control Backend Server (Node.js) ===');
  console.log('');
  
  try {
    // Check if dependencies are installed
    if (!checkDependencies()) {
      console.log('Installing missing dependencies...');
      await installDependencies();
    }
    
    // Start the server
    startServer();
    
  } catch (error) {
    console.log('Please install dependencies manually:');
    console.log('npm install');
    process.exit(1);
  }
}

if (require.main === module) {
  main();
}